components/compositing/compositor.rs

```diff
@@ -1781,77 +1810,82 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
 
-                let x = rect.origin.x as i32;
-                // We need to convert to the bottom-left origin coordinate
-                // system used by OpenGL
-                let y = (size.height as f32 - rect.origin.y - rect.size.height) as i32;
-                let w = rect.size.width as u32;
-                let h = rect.size.height as u32;
+            let x = rect.origin.x as i32;
+            // We need to convert to the bottom-left origin coordinate
+            // system used by OpenGL
+            let y = (size.height as f32 - rect.origin.y - rect.size.height) as i32;
+            let w = rect.size.width as u32;
+            let h = rect.size.height as u32;
 
-                (x, y, w, h)
-            },
-            None => (0, 0, size.width, size.height),
+            (x, y, w, h)
+        } else {
+            (0, 0, size.width, size.height)
         };
 
-        let rv = match target {
-            CompositeTarget::Window => None,
-            #[cfg(feature = "gl")]
-            CompositeTarget::WindowAndPng => {
-                let img = gl::draw_img(
-                    &*self.webrender_gl,
-                    rt_info,
-                    x,
-                    y,
-                    FramebufferUintLength::new(width),
-                    FramebufferUintLength::new(height),
-                );
-                Some(Image {
-                    width: img.width(),
-                    height: img.height(),
-                    format: PixelFormat::RGB8,
-                    bytes: ipc::IpcSharedMemory::from_bytes(&*img),
-                    id: None,
-                    cors_status: CorsStatus::Safe,
-                })
-            },
-            #[cfg(feature = "gl")]
-            CompositeTarget::PngFile => {
-                let gl = &*self.webrender_gl;
-                profile(
-                    ProfilerCategory::ImageSaving,
-                    None,
-                    self.time_profiler_chan.clone(),
-                    || match self.output_file.as_ref() {
-                        Some(path) => match File::create(path) {
-                            Ok(mut file) => {
-                                let img = gl::draw_img(
-                                    gl,
-                                    rt_info,
-                                    x,
-                                    y,
-                                    FramebufferUintLength::new(width),
-                                    FramebufferUintLength::new(height),
-                                );
-                                let dynamic_image = DynamicImage::ImageRgb8(img);
-                                if let Err(e) = dynamic_image.write_to(&mut file, ImageFormat::Png)
-                                {
-                                    error!("Failed to save {} ({}).", path, e);
-                                }
+        cfg_if! {
+            if #[cfg(feature = "gl")] {
```

Comment:
I honestly suspect that the `gl` feature might be a vestige of a time before WebRender.

Reply:
yeah, i think you’re right; removed in ae432a9284c75b81cd14e03fccc3164c2da64e8f